exes/ethercat/inc/public/tfc/ec/devices/schneider/atv320/dbus-iface.hpp

```diff
@@ -248,10 +257,33 @@ struct controller {
     }
   }
 
+  void on_positive_limit_switch(bool new_v) {
+    positive_limit_value_ = new_v;
+    logger_.trace("New positive limit switch value: {}", new_v);
+    if (pos_.positive_limit_switch().has_value() && pos_.homing_sensor().has_value()) {
+      if (pos_.positive_limit_switch().value().connection() == pos_.homing_sensor().value().connection()) {
+        homing_complete_.notify_all();
+      }
+    }
+    on_limit_switch(new_v, motor::errors::err_enum::positioning_positive_limit_reached);
+  }
```

Comment:
I am kind of surprised this works. I would think `on_limit_switch` would be called before the callback of `notify_all` could it be this code is never "hit" in testing cause the homing_sensor connects first to the signal or do you have more insight here?

Reply:
my thought was that `on_limit_switch` is not a async initiator but executes its logic as soon as called. but notify_all probably just schedules work onto an execution context.

I would have thought there should be a return after `homing_complete_.notify_all();`

Reply:
First paragraph answer: on_limit_switch is also scheduling work onto the execution context, therefore, it will initiate after the homing_complete_ token is scheduled.

Second paragraph answer: Yes I skipped the return for a reason, if someone calls `move(0)` it should stop at the limit switch even though it is connected to the same input as homing sensor. In that call stack no one is listening to homing_complete_ condition so that one will not activate any token. 


Is this logic too complex or do you have suggestions on making this more clear?